### DISCUSSION TITLE
How to process data when SQL is not an option?

### DISCUSSION THREAD

**Discussion by Riduidel** [dataform-co/dataform#1545] [Q&A] [answered]
In my use case, there are some things that I don't envision doing with SQL (typically extracting the git history of a repository to generate an history table). What would be the best way, when using GCP Dataform, to do that?

**Comment by Riduidel** [comment]
I guess I can use remote functions ? (https://cloud.google.com/bigquery/docs/remote-functions#use_a_remote_function_in_a_query)

**Reply by BenBirt** [reply, marked as answer]
Dataform only supports SQL.

That said: yes, if I wanted to do this, I would probably use a remote function.

**Comment by Riduidel** [comment]
If so, do you think it would be a good or a bad idea to have the cloud function code in the GitHub repository I use for Dataform? Or should I create a secondary repository just for that cloud function?

**Reply by BenBirt** [reply]
Personally I'd keep everything in one place - but I love a monorepo!
